SIGSEGV: segmentation violation on shell source when environment variable isn't provided.

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Current Behavior



SIGSEGV thrown when env variables are specified in a shell but not provided.

Our github action has these env variables but if I test locally then my aws cli is authenticated by a different mechanism and doesn't have these variables

```
sources:
  dockerBackend:
    name: "Docker Image Published on Registry"
    kind: shell
    spec:
      environments:
          - name: AWS_DEFAULT_REGION
          - name: AWS_REGION
          - name: AWS_ACCESS_KEY_ID
          - name: AWS_SECRET_ACCESS_KEY
          - name: AWS_SESSION_TOKEN
      command: |-
        aws ecr describe-images \
        --repository-name repo/imagename \
        --query "sort_by(imageDetails[?imageTags[0] && starts_with(imageTags[0], 'release-')], &imagePushedAt)[-1] | imageTags[0]" \
        --output json | tr -d \"
```

I source snippet like so yields the following error:

```
SCM repository retrieved: 1


++++++++++++++++++
+ AUTO DISCOVERY +
++++++++++++++++++



++++++++++++
+ PIPELINE +
++++++++++++



############################################
# BUMP ARTISIGHT-APP HELM CHART IMAGE TAGS #
############################################

source: source#dockerBackend
--------------------
ERROR: environment variable "AWS_DEFAULT_REGION" not found, skipping
ERROR: environment variable "AWS_REGION" not found, skipping
ERROR: environment variable "AWS_ACCESS_KEY_ID" not found, skipping
ERROR: environment variable "AWS_SECRET_ACCESS_KEY" not found, skipping
ERROR: environment variable "AWS_SESSION_TOKEN" not found, skipping
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x106b0e200]

goroutine 10 [running]:
github.com/updatecli/updatecli/pkg/plugins/resources/shell.Environment.String(...)
        /home/runner/work/updatecli/updatecli/pkg/plugins/resources/shell/environment.go:48
github.com/updatecli/updatecli/pkg/plugins/resources/shell.Environments.ToStringSlice({0x14000954d20, 0x6, 0x0?})
        /home/runner/work/updatecli/updatecli/pkg/plugins/resources/shell/environments.go:17 +0xb0
github.com/updatecli/updatecli/pkg/plugins/resources/shell.(*Shell).Source(0x140004727e0, {0x14000072004, 0x32}, 0x14000417860)
        /home/runner/work/updatecli/updatecli/pkg/plugins/resources/shell/source.go:48 +0x200
github.com/updatecli/updatecli/pkg/core/pipeline/source.(*Source).Run(0x14000417860)
        /home/runner/work/updatecli/updatecli/pkg/core/pipeline/source/main.go:90 +0x530
github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).RunSource(0x14000bf2b40, {0x14000ebea10, 0xd})
        /home/runner/work/updatecli/updatecli/pkg/core/pipeline/sources.go:14 +0x13c
github.com/updatecli/updatecli/pkg/core/pipeline.(*Pipeline).runFlowCallback(0x14000bf2b40, 0x14000428690, {0x140013ddf50, 0x14}, {0x14000791500, 0x1, 0x1?})
        /home/runner/work/updatecli/updatecli/pkg/core/pipeline/main.go:324 +0xa8c
github.com/heimdalr/dag.(*DAG).DescendantsFlow.func1({0x140013ddf50, 0x14})
        /home/runner/go/pkg/mod/github.com/heimdalr/dag@v1.5.0/dag.go:997 +0x140
created by github.com/heimdalr/dag.(*DAG).DescendantsFlow in goroutine 1
        /home/runner/go/pkg/mod/github.com/heimdalr/dag@v1.5.0/dag.go:983 +0x3ec
```

### Expected Behavior

It should not throw an error when the env values don't exist.

### Steps To Reproduce

1. create a source using shell and list env names but don't provide them in the underlying shell
2. call update cli and observe the SIGSEGV



### Environment

```markdown
- OS:
- updatecli:
```

### Pipeline Graph

_No response_

### Anything else?

_No response_

--- a/pkg/plugins/resources/shell/environment.go
+++ b/pkg/plugins/resources/shell/environment.go
@@ -45,6 +45,9 @@ type Environment struct {
 }
 
 func (e Environment) String() string {
+	if e.Value == nil {
+		return e.Name + "="
+	}
 	return fmt.Sprintf("%s=%s", e.Name, *e.Value)
 }
 

